### PR TITLE
osrdyne, gateway, editoast: switch docker images to alpine

### LINKED ISSUE #8708: Change base image for built containers
We should use lighter images when building.

- Use `FROM scratch` for `gateway` and `osrdyne`
- Switch to alpine for the following images: 
  - `editoast`
  - for the build step of `front-nginx`
  - `core` (both gradle and temurin)

### PR DESCRIPTION
**Pull request by ghost**
Switching osrdyne, gateway and editoast to alpine base image.

Closes [#8708](https://github.com/OpenRailAssociation/osrd/issues/8708)

### CODE REVIEW

**Review by woshilapin — APPROVED** [review]
I love it! ❤️

**Comment by ghost** [comment]
@woshilapin : Since you're an editoast maintainer, could you refresh your approval ?

File: editoast/Dockerfile [hunk 1]
```diff
@@ -16,8 +16,9 @@ RUN cargo chef prepare --recipe-path recipe.json
 ######################
 FROM chef as base_builder
 # geos crate requirement
-RUN apt update -yqq && apt install -yqq --no-install-recommends libgeos-dev mold
+RUN apk add --no-cache build-base openssl-dev mold libpq-dev geos-dev geos
```
Comment (Khoyo) [inline editoast/Dockerfile]: Do we need geos-dev and geos?

File: editoast/Dockerfile [hunk 2]
```diff
@@ -44,10 +44,9 @@ COPY . .
 FROM base_builder AS run_builder
 COPY --from=planner /editoast/recipe.json recipe.json
 COPY --from=planner /editoast/editoast_derive/ editoast_derive
-ENV RUSTFLAGS="-C link-arg=-fuse-ld=mold"
 ARG CARGO_PROFILE=release
 ARG CARGO_FEATURES
-RUN --mount=type=cache,target=/usr/local/cargo/registry \
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
```
Comment (Khoyo) [inline editoast/Dockerfile:47]: Spurious spaces

Comment (emersion) [inline editoast/Dockerfile:47]: Nit: trailing spaces

**Review by Khoyo — APPROVED** [review]
Otherwise LGTM

**Review by emersion — APPROVED** [review]
Apart from the nit LGTM!

**Review by leovalais — APPROVED** [review]
LGTM (if it builds it's fine isn't it? :eyes:)

The tests are failing though.

**Review by woshilapin — APPROVED** [review]
Looks still good. I'm a bit like Léo, if tests and CI pass, it's probably fine 🥇

**Pull request merged**